cli/flox-rust-sdk/src/models/manifest.rs

```diff
@@ -1788,63 +1839,84 @@ pub(super) mod test {
             [install]
             python3 = { pkg-path = "python3", version = "3.11.6" }
 
-            # Set environment variables in the `[vars]` section. These variables may not
-            # reference one another, and are added to the environment without first
-            # expanding them. They are available for use in the `[profile]` and `[hook]`
-            # scripts.
+
+            ## Environment Variables ---------------------------------------------
+            ##  ... available for use in the activated environment
+            ##      as well as [hook], [profile] scripts and [services] below.
+            ## -------------------------------------------------------------------
             [vars]
-            # message = "Howdy"
+            # INTRO_MESSAGE = "It's gettin' Flox in here"
+
 
-            # The `hook.on-activate` script is run by the *bash* shell immediately upon
-            # activating an environment, and will not be invoked if Flox detects that the
-            # environment has previously been activated. Variables set by the script will
-            # be inherited by `[profile]` scripts defined below. Note that any stdout
-            # generated by the script will be redirected to stderr.
+            ## Activation Hook ---------------------------------------------------
+            ##  ... run by _bash_ shell when you run 'flox activate'.
+            ## -------------------------------------------------------------------
             [hook]
             # on-activate = '''
-            #   # Set variables, create files and directories
-            #   venv_dir="$(mktemp -d)"
-            #   export venv_dir
-            #
-            #   # Perform initialization steps, e.g. create a python venv
-            #   python -m venv "$venv_dir"
+            #   # -> Set variables, create files and directories
+            #   # -> Perform initialization steps, e.g. create a python venv
+            #   # -> Useful environment variables:
+            #   #      - FLOX_ENV_PROJECT=/home/user/example
+            #   #      - FLOX_ENV=/home/user/example/.flox/run
+            #   #      - FLOX_ENV_CACHE=/home/user/example/.flox/cache
             # '''
 
-            # Scripts defined in the `[profile]` section are *sourced* by *your shell* and
-            # inherit environment variables set in the `[vars]` section and by `[hook]` scripts.
-            # The `profile.common` script is sourced by all shells and special care should be
-            # taken to ensure compatibility with all shells, after which exactly one of
-            # `profile.{bash,fish,tcsh,zsh}` is sourced by the corresponding shell.
+
+            ## Profile script ----------------------------------------------------
+            ## ... sourced by _your shell_ when you run 'flox activate'.
+            ## -------------------------------------------------------------------
             [profile]
             # common = '''
```

Comment:
I think errors resulting from this are not entirely obvious. I think a short comment highlighting this would be helpful.
We might also place the shell specific ones above `common` to nudge people writing specific scripts.


```suggestion
            ## Sourced by all shells, shell specific syntax should be added below
            # common = '''
```

Reply:
One thing we skip in this iteration of manifest is how to explain the order of activation and what is the difference between `hook.on-activate` and `profile.*` with one sentence. If you have any ideas how to showcase this in few lines that would be awesome.